websites/F/FRAnime/presence.ts

```diff
@@ -0,0 +1,132 @@
+	Selectors = {
+		watching:
+			'div[class="aspect-video border-[1.5px] border-gray-700 rounded-lg flex w-full justify-center align-items-center"]',
+		players: {
+			vk: "#video_player > div > div.videoplayer_media > video",
+			sibnetSendvid: "#video_html5_wrapper_html5_api",
+			myvi: "#player > video.player-video",
+			FRAnime: 'video[class="art-video"]',
+		},
+	},
+	// Define the strings that will be used
+	presenceStrings = {
+		browsing: "🗺️ Parcours le catalogue",
+		branding: "🍜 Animes gratuit & sans pub",
+		websiteName: "FRAnime.fr",
+		onPage: "📺 Sur la page",
+		watchOn: "🔗 Regarder sur FRAnime.fr",
+	},
+	// Define the animeBrowsingTimestamp that will be used for the elapsed time
+	animeBrowsingTimestamp = Math.floor(Date.now() / 1000);
+
+// Main function
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {},
+		{ href } = window.location;
+
+	// Home screen presence
+	presenceData.details = presenceStrings.browsing;
+	presenceData.state = presenceStrings.branding;
+	presenceData.smallImageText = presenceStrings.websiteName;
+	presenceData.largeImageKey = playerAssets.logoBg;
+	presenceData.smallImageKey = playerAssets.search;
+	presenceData.startTimestamp = animeBrowsingTimestamp;
+	presenceData.buttons = [
+		{
+			label: `🔗 ${presenceStrings.websiteName}`,
+			url: "https://franime.fr",
+		},
+	];
+
+	// Detect if the client is on an anime page
+	if (href.split("/").includes("anime")) {
+		// Get the scriptElement that contains the episode info
+		const scriptElement = document.querySelector(
+				'script[type="application/ld+json"]'
+			) as HTMLScriptElement,
+			jsonLD = scriptElement
+				? JSON.parse(scriptElement.textContent || "")
+				: null;
+
+		// Set the anime page presence
+		presenceData.details = `${presenceStrings.onPage} ${jsonLD.name}`;
+		presenceData.state = presenceStrings.branding;
+		presenceData.smallImageText = presenceStrings.websiteName;
+		presenceData.largeImageKey = jsonLD.thumbnailUrl;
+		presenceData.smallImageKey = playerAssets.search;
+		presenceData.startTimestamp = animeBrowsingTimestamp;
+		presenceData.buttons = [
+			{
+				label: presenceStrings.watchOn,
+				url: href,
+			},
+		];
+
+		// Detect if the client is on the watching page
+		if (document.querySelector(Selectors.watching)) {
+			// Set the episode info as presence
+			presenceData.details = `📺 ${jsonLD.name} - S${
+				href.split("=")[1].split("&")[0]
+			}, EP${href.split("=")[2].split("&")[0]}`;
+			presenceData.state = presenceStrings.branding;
+			presenceData.smallImageText = presenceStrings.websiteName;
+			presenceData.smallImageKey = playerAssets.pause;
+			presenceData.buttons = [
+				{
+					label: presenceStrings.watchOn,
+					url: href,
+				},
+			];
+
+			// The anime is loading, so we delete the timestamp
+			delete presenceData.startTimestamp;
+			delete presenceData.endTimestamp;
+
+			// Detect if a player is loaded
+		} else if (document.querySelectorAll("video")) {
```

Comment:
As this uses `querySelectorAll`, this condition is always true. Did you mean `querySelector`?

Reply:
querySelectorAll is isued to detect the player box. Depending on players it might not detect if using querySelector

Reply:
[`querySelectorAll`](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) always returns a NodeList, which is truthy even if it is empty.
I assume that the if statement is checking for a video to exist before continuing, thus should use [`querySelector`](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector) which returns null (falsy) if it doesn't exist.